src/nvim/eval/funcs.c

```diff
@@ -10224,9 +10223,12 @@ static void f_stdioopen(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     semsg(e_stdiochan2, error);
   }
 
-
   rettv->vval.v_number = (varnumber_T)id;
   rettv->v_type = VAR_NUMBER;
+
+  if (!tv_dict_get_callback(opts, S_LEN("on_print"), &on_print)) {
+    // Not a problem
```

Comment:
it returns `false` on error, not for missing callback. so this should be next to `"on_stdin"` check and also return on failure.

Reply:
Woops, I should have read the docs instead of trying to guess. Thanks for catching this problem!